api/src/license_serializer.rs

```diff
@@ -0,0 +1,46 @@
+#![allow(arithmetic_overflow)]
+
+use std::num::Wrapping;
+
+pub trait LicenseKeySerializer {
+    fn hash(&self, seed: &[u8], magic: &[u8]) -> u8;
+    fn deserialize_key(&self, key: String) -> Vec<u8>;
+    fn serialize_key(&self, key: &[u8]) -> String;
+}
+
+pub struct DefaultLicenseKeySerializer {}
+
+impl LicenseKeySerializer for DefaultLicenseKeySerializer {
+    #[inline(always)]
+    fn hash(&self, seed: &[u8], magic: &[u8]) -> u8 {
+        let mut hash: Wrapping<u8> = Wrapping(0);
+
+        for &x in magic.iter() {
```

Comment:
```rust
let seed_sum = seed.sum<Wrapping<u8>>();  // however using u64 and then &0xff should be sufficient
magic.iter().fold(Wrapping(0u8), |(hash, &mb)|{
    match mb {
        3|7 => hash * x,
        k if k % 2 == 0 => hash + seed_sum,
        l => hash ^ l
    }
}).0
```

Reply:
thank, I take something from this, but I will keep this simple as example.